src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentDescriptorCollectionProvider.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+
+namespace Microsoft.AspNet.Mvc.ViewComponents
+{
+    /// <summary>
+    /// A default implementation of <see cref="IViewComponentDescriptorCollectionProvider"/>
+    /// </summary>
+    public class DefaultViewComponentDescriptorCollectionProvider : IViewComponentDescriptorCollectionProvider
+    {
+        private readonly IViewComponentDescriptorProvider _descriptorProvider;
+        private ViewComponentDescriptorCollection _viewComponents;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DefaultViewComponentDescriptorCollectionProvider"/>.
+        /// </summary>
+        /// <param name="descriptorProvider">The <see cref="IViewComponentDescriptorProvider"/>.</param>
+        public DefaultViewComponentDescriptorCollectionProvider(IViewComponentDescriptorProvider descriptorProvider)
+        {
+            _descriptorProvider = descriptorProvider;
+        }
+
+        /// <inheritdoc />
+        public ViewComponentDescriptorCollection ViewComponents
+        {
+            get
+            {
+                if (_viewComponents == null)
+                {
+                    _viewComponents = GetViewComponents();
+                }
```

Comment:
won't the singleton only be instantiated if something references the `IViewComponentDescriptorCollectionProvider` service?  and is that something almost-always going to use this property?  if so, lazy-init isn't buying us much.